library/alloc/src/collections/btree/node.rs

```diff
@@ -1610,7 +1609,7 @@ unsafe fn slice_insert<T>(slice: &mut [T], idx: usize, val: T) {
     unsafe {
         let len = slice.len();
         let slice_ptr = slice.as_mut_ptr();
-        ptr::copy(slice_ptr.add(idx), slice_ptr.add(idx + 1), len - idx);
+        ptr::copy(slice_ptr.add(idx), slice_ptr.add(idx + 1), len - idx - 1);
```

Comment:
Maybe add a comment explaining that this is the *new* `len` and hence we have to subtract one more.